### PR TITLE
CNV1610 - Installing VirtIO drivers for Windows VMs

### PR DESCRIPTION
**Pull request by aburdenthehand**
Two assemblies covering the user stories for installing virtio drivers to Win VMs - migrated from 1.4 content:
- Installing the drivers as part of an existing (Windows already installed on VM) VM
- Installing the drivers on a VM during installation of Windows on VM
Heavy content duplication between the two assemblies.

### CODE REVIEW

**Comment by vikram-redhat** [comment]
@openshift/team-documentation PTAL for peer review

File: modules/cnv-installing-virtio-drivers-installing-windows.adoc
Comment (kalexand-rh) [inline modules/cnv-installing-virtio-drivers-installing-windows.adoc]: This sounds like a UI element, so it should be enclosed in asterisks.

File: modules/cnv-installing-virtio-drivers-installing-windows.adoc
Comment (kalexand-rh) [inline modules/cnv-installing-virtio-drivers-installing-windows.adoc]: s/recognised/recognized

**Review by kalexand-rh — COMMENTED** [review]
I have a couple of requests, but this largely LGTM. I'm glad that you were able to reuse so much content!

File: cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc
Comment (petrkotas) [reply cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc]: It can be done by adding the windows image as a CDROM. Usually as a SATA bus.
```yaml
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachineInstance
metadata:
  labels:
    special: vmi-windows
  name: vmi-windows
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: sata
        name: pvcdisk
      - cdrom:
          bus: sata
        name: wininstall
  volumes:
  - name: pvcdisk
    persistentVolumeClaim:
      claimName: disk-windows
  - name: wininstall
    persistentVolumeClaim:
      claimName: disk-windows-install
```

Do you have a guide somewhere you can reference? In my opinion it would be better than just adding it here.

File: modules/cnv-adding-virtio-drivers-vm-yaml.adoc
Comment (petrkotas) [reply modules/cnv-adding-virtio-drivers-vm-yaml.adoc]: It is part of the Red Hat container catalogue. It took me a while to scan it. The information from the first line was not clearly visible for me. Although the information is present, it should be more visible. In my opinion, Red Hat catalogue should be a requirement, which would place it more in front of the eyes.

File: modules/cnv-adding-virtio-drivers-vm-yaml.adoc
Comment (petrkotas) [reply modules/cnv-adding-virtio-drivers-vm-yaml.adoc]: The VM can be started after the virtio-win is added to the VM specification. 
This would make more sense for me, if it were written as a step 2, telling me to start the VM and wait for the boot.

File: modules/cnv-installing-virtio-drivers-existing-windows.adoc
Comment (petrkotas) [reply modules/cnv-installing-virtio-drivers-existing-windows.adoc]: They are located on the mounted CDROM. @aburdenthehand would you please add the reference here? The guide is long and one can be easily lost.

File: modules/cnv-understanding-virtio-drivers.adoc
Comment (petrkotas) [reply modules/cnv-understanding-virtio-drivers.adoc]: @aburdenthehand I agree with @kalexand-rh the link is https://access.redhat.com/containers/#/product/5be1983a5a13463a3e1d8ef4. It will make reading the guide much easier.

**Review by petrkotas — CHANGES_REQUESTED** [review]
@aburdenthehand thank you for your effort with the guide. You did a great job. May I offer you a few comments, which would make the guide better? 

One note in general, the guide would look much better with screenshots. Similar to the upstream guide https://kubevirt.io/user-guide/docs/latest/creating-virtual-machines/virtio-win.html. It allows our customers to get much better grasp of what is going on. Not everyone is fluent with the windows.

Btw, I am the one responsible for the upstream guide.

File: modules/cnv-supported-virtio-drivers.adoc
Comment (aburdenthehand) [reply modules/cnv-supported-virtio-drivers.adoc]: They're driver names, but only viewed in GUI so should be okay to leave as is.

File: modules/cnv-supported-virtio-drivers.adoc
Comment (aburdenthehand) [reply modules/cnv-supported-virtio-drivers.adoc]: Depending on the Windows install, it might display as the listed driver name or it might not. If the latter, then it's likely in the *Other devices* under X controller/device.
Updated to read: "Sometimes displays as a..."

File: modules/cnv-adding-virtio-drivers-vm-yaml.adoc
Comment (aburdenthehand) [reply modules/cnv-adding-virtio-drivers-vm-yaml.adoc]: I've made a couple of changes:
* changed all mentions of the 'virt-win' container disk to include the 'cnv-tech-preview' namespace. Although the namespace will likely change soon, it's needed for the yaml and this keeps it consistent throughout. I think this change,in context, makes t easier to follow.
* Also added the suggested prerequisite. It's a bit of an odd one, because it's not necessary, since the container will be downloaded if it's not in the cluster, but at 150mb I guess it's big enough to speed up the procedure.
I've posted the build here: file:///home/aburden/git/openshift-docs/_preview/openshift-enterprise/cnv1610-virtio-drivers/cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-existing-windows-vm.html#cnv-adding-virtio-drivers-vm-yaml_cnv-installing-virtio-drivers-on-existing-windows-vm
(There's something screwy with the css when I transfer to that fileserver. It doesn't render like that locally, nor will it once it's merged)

File: modules/cnv-adding-virtio-drivers-vm-yaml.adoc
Comment (aburdenthehand) [reply modules/cnv-adding-virtio-drivers-vm-yaml.adoc]: As suggested, the admonition has been folded into a second step, which reads:
. The disk is available once the virtual machine has started:
** If you add the container disk to a running virtual machine, use
`oc apply -f <vm.yaml>` in the CLI or reboot the virtual machine for the changes
to take effect.
** If the virtual machine is not running, use `virtctl start <vm>`.

File: modules/cnv-installing-virtio-drivers-existing-windows.adoc
Comment (aburdenthehand) [reply modules/cnv-installing-virtio-drivers-existing-windows.adoc]: The first sentence in this step now reads:
Click *Browse my computer for driver software* and browse to the attached SATA CD drive, where the VirtIO drivers are located.

File: modules/cnv-removing-virtio-disk-from-vm.adoc
Comment (aburdenthehand) [reply modules/cnv-removing-virtio-disk-from-vm.adoc]: I've changed it so that is just 'edit the config', and included the $ oc edit vmi <vmi> command before the config snippet.
When the Web Console content for 1.4 (covers start, stop, restart, edit, etc) is available, I can xref to them as 'See Also' in the assemblies

File: cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc
Comment (aburdenthehand) [reply cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc]: @petrkotas To get the Windows installation media in the pvc, do we use the same concept as documented here?
https://docs.openshift.com/container-platform/3.11/cnv_users_guide/cnv_users_guide.html#importvm-datavolumes
If so, I can xref to that once it has been migrated (in progress)

**Comment by aburdenthehand** [comment]
@petrkotas @kalexand-rh - Thank you both for all the feedback. I've pushed all changes and responded to everything else in comments. 
Petr, re: screenshots. I do think that screenshots can be effective but I'm very reluctant to add them here, especially since there's no definitive version of Windows installation. I believe OpenShift docs is also very careful with using screenshots, since they easily become a maintenance burden and create accessibility issues. 
And thanks too for your work on the upstream doc! It was very useful when working on this for the previous release.

File: cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc
Comment (petrkotas) [reply cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc]: @aburdenthehand that is a good find. Yes you can do it that way. You can also populate the persistent volume manually, but in my opinion we should not point that in the guide.

**Review by petrkotas — APPROVED** [review]
Hi @aburdenthehand thanks for the updates. The guide looks good.

File: cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc
Comment (aburdenthehand) [reply cnv/cnv_users_guide/cnv-installing-virtio-drivers-on-new-windows-vm.adoc]: @petrkotas - I've linked that prerequisite to the `importvm-datavolumes` content linked in the earlier comment. That content is still being updated+migrated but I've included the anchor of what it will be called.

@kalexand-rh I believe this is the only outstanding issue with this PR. If this looks okay then it's ready to merge.

**Pull request merged**
